courses/models.py

```diff
@@ -287,3 +288,30 @@ def satisfying_courses(self):
         return Course.objects.all()\
             .exclude(id__in=self.overrides.all())\
             .filter(Q(department__in=self.departments.all(), semester=self.semester) | Q(id__in=self.courses.all()))
+
+
+"""
+3rd-Party API
+"""
+
+PCA_REGISTRATION = 'PCA_REGISTRATION'
+
+
+class APIPrivelege(models.Model):
+    """
+    Describes a type of access privelege that an API key can have.
+    """
+    code = models.CharField(max_length=255, unique=True)
+    description = models.TextField(blank=True)
+
+
+class APIKey(models.Model):
```

Comment:
Does it make more sense to build our own API key, or just use an existing package like [djangorestframework-api-key](https://github.com/florimondmanca/djangorestframework-api-key)?

Reply:
This looks like it makes sense. I'll see if I have time to plug this in tonight.

Reply:
After talking in person, Armaan and I came to the agreement that the linked library 1) doesn't implement a lot of functionality that we couldn't easily implement ourselves and 2) doesn't seem super popular and might be at risk of abandonment. Reason 2 tends to outweigh the concerns from reason 1 of reinventing the wheel in this instance.